src/controllers/preset/editPresetController.ts

```diff
@@ -9,11 +22,23 @@ const editPresetController = async (
     interface Preset {
       id?: number;
       name?: string;
-      type?: string;
+      price?: number;
+      color?: string;
+      type?: PresetTypes;
+      category?: PresetCategories;
+      isDisabled?: boolean;
     }
     const { id }: Preset = req.params;
-    const { name, type }: Preset = req.query;
-    const preset = await editPresetHandler(id, name, type);
+    const { name, price, color, type, category, isDisabled }: Preset =
+      req.query;
+    const preset = await editPresetHandler(id, {
```

Comment:
Why the id is passed like this? Its just a doubt i have

Reply:
id is a parameter that should always be present but the data you want to update isn't, that way i can check for missing data separately